api/src/pcapi/connectors/acceslibre.py

```diff
@@ -263,11 +263,52 @@ def extract_street_name(address: str | None = None, city: str | None = None, pos
     return ""
 
 
+def address_match(
+    acceslibre_address: str | None = None,
+    acceslibre_city: str | None = None,
+    acceslibre_postal_code: str | None = None,
+    acceslibre_ban_id: str | None = None,
+    venue_address: str | None = None,
+    venue_city: str | None = None,
+    venue_postal_code: str | None = None,
+    venue_ban_id: str | None = None,
+) -> bool:
+    # If banIds are matching, we return True
+    ban_ids_match = venue_ban_id and venue_ban_id == acceslibre_ban_id
+    # If cities / postal codes are given, we must check that they do not mismatching, else we return False
+    cities_mismatch = (venue_city and venue_city != acceslibre_city) or (
+        venue_postal_code and venue_postal_code != acceslibre_postal_code
+    )
+    # if cities / postal codes are defined and are matching, we check for the matching of the address (street + number)
+    addresses_match = (
+        venue_address and acceslibre_address and fuzz.ratio(acceslibre_address, venue_address) > ADDRESS_MATCHING_RATIO
+    )
+
+    if ban_ids_match or (not cities_mismatch and addresses_match):
+        return True
+    return False
+
+
+def name_match(acceslibre_name: str, venue_name: str, venue_public_name: str) -> bool:
+    if (  # pylint: disable=too-many-boolean-expressions
+        acceslibre_name in venue_name
+        or venue_name in acceslibre_name
+        or venue_public_name in acceslibre_name
+        or acceslibre_name in venue_public_name
+        or fuzz.ratio(acceslibre_name, venue_name) > NAME_MATCHING_RATIO
+        or fuzz.ratio(acceslibre_name, venue_public_name) > NAME_MATCHING_RATIO
+    ):
+        return True
+    return False
```

Comment:
Ok pour le déplacement de la condition vers une fonction dédiée.

Je pense à quelque chose à propos des `in` pour le matching : par exemple le `acceslibre_name in venue_name`, si on a 2 venues `"Mairie d'Aize"` et `"Mairie d'Aizenay"` et un acceslibre_name à `"Mairie d'Aize"` on peut matcher `"Mairie d'Aizenay"` --> je me demande si on pourrait pas améliorer cette partie là (est-ce que le `in` est vraiment nécessaire et ne pourrait pas être remplacé par autre chose par exemple ?)

Reply:
Ok en combinaison avec le address_match on resoud les éventuels résultats multiples obtenus sur le name_match

Reply:
Peut-être quelque chose comme ça :
`normalize_name(acceslibre_name) == normalize_name(venue_name)`
avec une fonction `normalize_name()` à définir, qui fera probablement les opérations `strip()` / `lower` / gestion de la ponctuation et autres si nécessaire

Reply:
Comme ça ne concerne pas le sujet de la PR je ferme ce commentaire, c'était une idée en passant, peut-être à regarder une autre fois :)

Reply:
Oui j'avais plusieurs cas où le nom chez nous c'était par exemple `CHRD` et côté acceslibre `CHRD - Centre d'Histoire de la Résistance et de la Déportation de Lyon`-> Le ratio sur le nom est bien inférieur à 50% pourtant c'est bien le même lieu. D'où cette condition qui améliorait bien le taux de match